### PR TITLE
[Dashboard] Simplify filters stickiness calculation

### LINKED ISSUE #35078: Filters blink when there are many lines of filters in an embedded context with headers=false
### Describe the bug

If you have many lines of filter on an interactive embedding context with headers = false, filters will blink

blink

### To Reproduce

1) use the full app embedding demo https://github.com/metabase/sso-examples/tree/master/app-embed-example, but ensure that you add headers=false to the embedding URL (https://www.metabase.com/docs/latest/embedding/interactive-embedding#header)
2) as an admin, log into metabase and build a dashboard with an x-ray and then add several filters till you get to a second line
3) then autheticate as a user via the embedding
4) go to the dashboard and scroll down, you'll see the filters blinking

### Expected behavior

filters shouldn't blink

### Logs

NA

### Information about your Metabase installation

```JSON
v47.x
```

### Severity

P2

### Additional context

NA

### PR DESCRIPTION
**Pull request by uladzimirdev**
Closes https://github.com/metabase/metabase/issues/35078

### Description

Rely on CSS instead of manual calculation for a sticky filters container on the dashboard.

previous PRs were about js solution (when support of `position: sticky` wasn't that great)
https://github.com/metabase/metabase/pull/17406
https://github.com/metabase/metabase/pull/21069
https://github.com/metabase/metabase/pull/21848
https://github.com/metabase/metabase/pull/23692
https://github.com/metabase/metabase/pull/24669

### How to verify

Describe the steps to verify that the changes are working as expected.

1. follow steps from the issue https://github.com/metabase/metabase/issues/35078 or
1. go to the dashboard, add some parameters, find "header" elements in the dev tools and remove it, scroll and see that filters contains do not blink

### Demo

https://github.com/metabase/metabase/assets/125459446/d310b89c-d4d0-48d2-8fdd-f7c8344dbd8a

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/dashboard/components/Dashboard/stickyParameters.js [hunk 1]
```diff
@@ -17,33 +17,8 @@ export const updateParametersWidgetStickiness = dashboard => {
...
-  if (!deviceShouldDisplayStickyFilters) {
-    return false;
-  }
-
-  const offsetTop = getOffsetTop(dashboard);
-
-  return getMainElement().scrollTop > offsetTop;
-};
-
-const getOffsetTop = dashboard => {
```
Comment (uladzimirdev) [inline frontend/src/metabase/dashboard/components/Dashboard/stickyParameters.js:45]: we rely on [css](https://caniuse.com/?search=position%3A%20sticky) nowadays

**Review by ranquild — APPROVED** [review]

**Comment by paoliniluis** [comment]
LGTM!

**Review by dragonsahead — APPROVED** [review]

**Review by npfitz — COMMENTED** [review]
Love this change, it's something that's been in the back of my mind since the last time I touched this haha. I'd love to verify that the stickyParameters behavior is covered in an e2e test somewhere. I just took a quick look at didn't see one, but if it is that's fantastic. I'm not sure we'll be able to rest this reliably with a unit test :/

**Comment by uladzimirdev** [comment]
@npfitz actually I found out that we need a sticky container always, except small screen and 6+ filters. This case is covered by unit tests

**Comment by npfitz** [comment]
That makes sense. I was just thinking of a test that scrolls to the bottom of a dashboard and ensures that the params are still there where they're supposed to be. Could be useful if we ever change the structure of the dashboard component moving forward.

**Comment by uladzimirdev** [comment]
to my mind we have exactly that unit test 😄 but I understand your idea

**Pull request merged**